drivers/spi/spi.c

```diff
@@ -85,3 +85,34 @@ int32_t spi_write_and_read(struct spi_desc *desc,
 {
 	return desc->platform_ops->write_and_read(desc, data, bytes_number);
 }
+
+/**
+ * @brief  Iterate over head list and send all spi messages
+ * @param desc - The SPI descriptor.
+ * @param head - First link of the list with messages to be sent
+ * @return SUCCESS in case of success, negativ error code otherwise.
+ */
+int32_t spi_transfer(struct spi_desc *desc,
+		     struct spi_msg_list *head)
+{
+	int32_t ret;
+
+	if (!desc || !desc->platform_ops)
+		return -EINVAL;
+
+	if (desc->platform_ops->transfer)
+		return desc->platform_ops->transfer(desc, head);
+
+	while (head) {
+		if (head->rx_buff != head->tx_buff)
```

Comment:
if these two pointers have to be the same why do you need two for the same thing ?

Reply:
This is in spi.c, if a platform don't defines a spi_transfer function, this will be used.
The check is because spi_write_and_read will be used and the buffer to be passed will be the same for write and read.